libs/SalesforceSDKCommon/SalesforceSDKCommon/Classes/Logger/SFLogger.m

```diff
@@ -67,82 +105,132 @@ -(void)setLogLevel:(SFLogLevel) logLevel {
 }
 - (void)log:(nonnull Class)cls message:(nonnull NSString *)message {
     [self.logger log:cls level:SFLogLevelDefault message:message];
+    
+    [self.logReceiver receiveWithLevel:SFLogLevelDefault cls:cls component:self.logger.componentName message:message];
 }
 
 - (void)log:(nonnull Class)cls format:(nonnull NSString *)format, ... {
     va_list args;
     va_start(args, format);
     [self.logger log:cls level:SFLogLevelDefault format:format args:args];
+    
+    NSString *formattedMessage = [[NSString alloc] initWithFormat:format arguments:args];
+    [self.logReceiver receiveWithLevel:SFLogLevelDefault cls:cls component:self.logger.componentName message:formattedMessage];
+    
     va_end(args);
 }
 
 - (void)log:(Class)cls level:(SFLogLevel)level message:(NSString *)message {
     [self.logger log:cls level:level message:message];
+    
+    [self.logReceiver receiveWithLevel:level cls:cls component:self.logger.componentName message:message];
 }
 
 - (void)log:(Class)cls level:(SFLogLevel)level format:(NSString *)format, ... {
     va_list args;
     va_start(args, format);
     [self.logger log:cls level:level format:format args:args];
+    
+    NSString *formattedMessage = [[NSString alloc] initWithFormat:format arguments:args];
+    [self.logReceiver receiveWithLevel:level cls:cls component:self.logger.componentName message:formattedMessage];
+    
     va_end(args);
 }
 
 - (void)log:(Class)cls level:(SFLogLevel)level format:(NSString *)format args:(va_list)args {
+    va_list args_copy;
+    va_copy(args_copy, args);
```

Comment:
Note: We have to copy the va_list since the logger will probably end it before we can parallel log it to the log receiver.

Reply:
If the logger ends the va_list, then the second call would result in a crash.  So, this local copy guards for that.